src/core/server.ts

```diff
@@ -34,6 +31,7 @@ export interface ServerOptions<T> {
   apolloConfig?: ApolloServerExpressConfig;
   authChecker?: AuthChecker<T>;
   autoGenerateFiles?: boolean;
+  connectionGetter?: (config: Config) => Connection | Promise<Connection>;
```

Comment:
I think I'd prefer this to either be a drop in `Connection` or for the input parameter to this function type be typeorm's `ConnectionOptions`.

Reply:
```suggestion
  connectionGetter?: (options: ConnectionOptions) => Connection | Promise<Connection>;
```
```suggestion
  connection?: Connection;
```

Reply:
`Connection` - in this case, there would be no way to reconnect in the case the server drops the DB connection, right?  Is that ok?
`ConnectionOptions` - `Server` already allows you to pass `ConnectionOptions`: https://github.com/goldcaddy77/warthog/pull/458/files#R67 .  Could you just use that?  I figured you needed access to the raw connection to be able to mutate it.